File: scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  def convert(n: Int): String =
    (n % 3 == 0, n % 5 == 0, n % 7 == 0) match
      case (true, true, true) => "PlingPlangPlong"
      case (true, false, false) => "Pling"
      case (false, true, false) => "Plang"
      case (false, false, true) => "Plong"
      case (true, true, false) => "PlingPlang"
      case (true, false, true) => "PlingPlong"
      case (false, true, true) => "PlangPlong"
      case (false, false, false) => n.toString()
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {
    fun convert(n: Int): String {
        if (n % 3 != 0 && n % 5 != 0 && n % 7 != 0) {
            return n.toString()
        } else {
            var str = ""
            if (n % 3 == 0) {
                str += "Pling"
            }
            if (n % 5 == 0) {
                str += "Plang"
            }
            if (n % 7 == 0) {
                str += "Plong"
            }
            return str
        }
    }
}
